File: Wms.Api/Dto/StockIn/StockInDetails/StockInDetailsDto.cs
namespace Wms.Api.Dto.StockIn.StockInDetails
{
    public class StockInDetailsDto
    {
        public Guid Id { get; set; }
        public required string Number { get; set; }
        public required string SellerInfo { get; set; }
        public required string Purchaser { get; set; }
        public DateTime DateOfPurchase { get; set; }
        public Guid WarehouseId { get; set; }
        public string WarehouseLabel { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; }
        public required string CreatedById { get; set; }
        public DateTime? ChangedAt { get; set; }
        public string? ChangedById { get; set; }

        public ICollection<StockInItemDetailsDto>? StockInItems { get; set; }
    }
}

File: Frontend/src/interfaces/v12/stockIn/stockInDetails/stockInDetailsDto.ts
import { guid } from "types/guid";
import { StockInItemDetailsDto } from "./stockInItemDetailsDto";

export interface StockInDetailsDto {
  id: guid;
  number: string;
  sellerInfo: string;
  purchaser: string;
  location: string;
  dateOfPurchase: string;
  warehouseId: guid;

  createdAt: string;
  createdById: string;
  changedAt?: string;
  changedById?: string;

  stockInItems?: StockInItemDetailsDto[];
}